### ISSUE TITLE
Fix S2589 FP: Deconstructing a tuple in foreach loop

### ISSUE THREAD

**Issue by dawust** [SonarSource/sonar-dotnet#7489]
### Description

When deconstructing a tuple in a foreach declaration S2589 is triggered on an unrelated variable.

### Repro steps

```
public static class Program
{
    public static void Main()
    {
        long? value;
        var hasValue = true;
        while (hasValue)
        {
            value = null;
            hasValue = value.HasValue;

            foreach (var (a,b) in new List<(int, int)>())
            {
                // triggers S2589 on hasValue
            }
            
            foreach (var x in new List<(int, int)>())
            {
                var (a, b) = x;
                // does not trigger S2589 on hasValue
            }
        }
    }
}
```

### Expected behavior

S2589 is not triggered

### Actual behavior

S2589 is triggered

### Related information

* C#/VB.NET Plugins version: SonarAnalyzer.CSharp 9.4.0.72892
* Rider 2022.3.3
* MSBuild / dotnet version: 7.0.203
* Operating System: Windows 11

**Comment by pavel-mikula-sonarsource** [comment]
Hi @dawust,

Thank you for reporting this case, I confirm it as False Positive.

We plan to reimplement the rule in Q3 and that should fix this problem.

**Comment by mary-georgiou-sonarsource** [comment]
Fixed by https://github.com/SonarSource/sonar-dotnet/pull/7750
